addons/hr_payroll_account/models/hr_payroll_account.py

```diff
@@ -0,0 +1,142 @@
+#-*- coding:utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+from odoo.tools import float_compare, float_is_zero
+
+
+class HrPayslipLine(models.Model):
+    _inherit = 'hr.payslip.line'
+
+    def _get_partner_id(self, credit_account):
+        """
+        Get partner_id of slip line to use in account_move_line
+        """
+        # use partner of salary rule or fallback on employee's address
+        partner = self.salary_rule_id.register_id.partner_id
+        partner_id = partner.id or self.slip_id.employee_id.address_home_id.id
+        if credit_account and partner or self.salary_rule_id.account_credit.internal_type in ('receivable', 'payable'):
+            return partner_id
+        elif partner or self.salary_rule_id.account_debit.internal_type in ('receivable', 'payable'):
+            return partner_id
+
+
+class HrPayslip(models.Model):
+    _inherit = 'hr.payslip'
+
+    date = fields.Date(string='Date Account', states={'draft': [('readonly', False)]}, readonly=True, help="Keep empty to use the period of the validation(Payslip) date.")
+    journal_id = fields.Many2one('account.journal', string='Salary Journal', states={'draft': [('readonly', False)]}, readonly=True, required=True, default=lambda self: self.env['account.journal'].search([('type', '=', 'general')], limit=1).id)
+    move_id = fields.Many2one('account.move', string='Accounting Entry', readonly=True, copy=False)
+
+    @api.model
+    def create(self, vals):
+        if 'journal_id' in self.env.context:
+            vals.update({'journal_id': self.env.context.get('journal_id')})
```

Comment:
directly we can use self.env.context['journal_id'] because already you check for journal_id in context..